src/typ.rs

```diff
@@ -3901,3 +4228,39 @@ pub fn tuple(elems: Vec<Arc<Type>>) -> Arc<Type> {
 pub fn fn_(args: Vec<Arc<Type>>, retrn: Arc<Type>) -> Arc<Type> {
     Arc::new(Type::Fn { retrn, args })
 }
+
+pub fn binary() -> Arc<Type> {
```

Comment:
We'll also want to insert the type constructor into the default type environment so that it can be referenced in user defined types: https://github.com/gleam-lang/gleam/blob/3e96b7714e862c36f93c38243ee748e6337d70a2/src/typ.rs#L443-L454

Reply:
Good point, will do.